dotCMS/src/main/java/com/dotmarketing/util/UserUtils.java

```diff
@@ -20,4 +20,16 @@ public static List getChallengeQuestionList() {
 
 		return ChallengeQuestionFactory.getChallengeQuestionList();
 	}
+
+	static final String ACCEPTABLE_PASSWORD_CHARS = "!#%+23456789:=?@ABCDEFGHJKLMNPRSTUVWXYZabcdefghijkmnopqrstuvwxyz";
+
+	/**
+	 * Generates a passwords that might be any random combination of 16 chars lenght
+	 * with any of these chars !#%+23456789:=?@ABCDEFGHJKLMNPRSTUVWXYZabcdefghijkmnopqrstuvwxyz
+	 * @return
+	 */
+	public static String generateSecurePassword(){
+	   	return RandomStringUtils.random(16, ACCEPTABLE_PASSWORD_CHARS.toCharArray());

```

Comment:
We will need at least 1 special char, one number, one uppercase and one lowercase in every generated password.